zebra-script/src/lib.rs

```diff
@@ -112,19 +143,21 @@ impl CachedFfiTransaction {
         self.transaction.inputs()
     }
 
-    /// Verify a script within a transaction given the corresponding
-    /// `transparent::Output` it is spending and the `ConsensusBranchId` of the block
-    /// containing the transaction.
-    ///
-    /// # Details
-    ///
-    /// The `input_index` corresponds to the index of the `TransparentInput` which in
-    /// `transaction` used to identify the `previous_output`.
-    pub fn is_valid(
-        &self,
-        branch_id: ConsensusBranchId,
-        (input_index, previous_output): (u32, transparent::Output),
-    ) -> Result<(), Error> {
+    /// Returns the outputs from previous transactions that match each input in the transaction
+    /// being verified.
+    pub fn all_previous_outputs(&self) -> &Vec<transparent::Output> {
+        &self.all_previous_outputs
+    }
+
+    /// Verify if the script in the input at `input_index` of a transaction correctly
+    /// spends the matching `transparent::Output` it refers to, with the `ConsensusBranchId`
```

Comment:
Nit: might be a good opportunity to add doc links :sweat_smile: 

```suggestion
    /// spends the matching [`transparent::Output`] it refers to, with the [`ConsensusBranchId`]
```

Reply:
Done in b36577315 / #3446